### PR TITLE
Implement an authentication system using social-auth and UserCreationForm

### PR DESCRIPTION
**Pull request by bemyXmas**
- Initial backend setup for login through Google, Twitter, Instagram, Facebook, and add login and sign in using google account button.
- Add register, sign-in, and sign-out using UserCreationForm.

### CODE REVIEW

File: accounts/views.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        user = authenticate(request, username=username, password=password)
+
+        if user is not None:
+            login(request, user)
+            return redirect('events:feed')
+        else:
+            messages.warning(request, 'Username or Password is incorrect.')
+
+    context = {}
+    return render(request, 'accounts/login.html', context)
```
Comment (nicenicegame) [inline accounts/views.py]: There is no context, so you don't need to pass it through the template.

**Comment by ttxking** [comment]
* When register with an invalid password, after fixing and create account the error message is shown on the login page. This should be fixed!
* Login with google work fine, No problem.
* After login, the event functionality works fine.
* Report system also works fines.
* Go to the page directly with URL work fines too.
* Join Jix button on index page should redirect us to register page!

File: accounts/templates/accounts/login.html [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                                <span class="input-group-text">Password</span>
+                            </div>
+                            <input type="password" name="password" class="form-control"/>
+                        </div>
+
+                        <div class="d-flex justify-content-center mt-3 login_container">
+                            <input class="btn login_btn btn-success my-4" type="submit" value="Login">
+                        </div>
+                        <a href="{% url "social:begin" "google-oauth2" %}">
+                            <div class='text-center'>
```
Comment (nicenicegame) [inline accounts/templates/accounts/login.html:30]: You should make the div with the 'text-center' class to wrap all of the icon and text, so you won't need to add a 'text-center' class to every div.

**Review by nicenicegame — COMMENTED** [review]
Some code can be refactored.

File: jix/urls.py [hunk 3]
```diff
@@ -16,10 +16,13 @@
...
 from . import views
+from django.conf.urls import url
 
 urlpatterns = [
     path('admin/', admin.site.urls),
     path('', views.index, name="index"),
     path('events/', include('events.urls')),
-    path('reports/', include('report.urls'))
+    path('reports/', include('report.urls')),
+    url('', include('social_django.urls', namespace='social')),
```
Comment (ttxking) [inline jix/urls.py]: this can be refactor to `path('', include('social_django.urls'), name='social'),`

**Review by ttxking — COMMENTED** [review]
You can do it later because `url(..)` is for an old version of Django.

File: jix/urls.py (hunk 3 above)
Comment (bemyXmas) [reply jix/urls.py]: ok

File: accounts/templates/accounts/login.html (hunk 2 above)
Comment (bemyXmas) [reply accounts/templates/accounts/login.html:30]: ok fixed it!

**Review by Kuea666 — COMMENTED** [review]
I pulled it to test and didn't find any problems.

File: jix/urls.py (hunk 3 above)
Comment (bemyXmas) [reply jix/urls.py]: fixed it!

File: accounts/views.py (hunk 1 above)
Comment (bemyXmas) [reply accounts/views.py]: fixed it!

**Review by ttxking — APPROVED** [review]

**Review by nicenicegame — APPROVED** [review]
I pulled and tested it, I think it okay now! Not found any errors!

**Review by Kuea666 — APPROVED** [review]
Gud

**Pull request merged**